### PR TITLE
usb: fix bulk endpoint configuration for high-speed capable device

### PR DESCRIPTION
**Pull request by jfischer-no**
In the current USB device support, the sizes of bulk endpoint
are mostly configure through Kconfig and do not care if a device
is high-speed capable. The information if a USB device controller
supports high-speed comes from devicetree. Add a Kconfig option to
map this information and configure bulk endpoint sizes
accordingly.

### CODE REVIEW

**Review by aurel32 — CHANGES_REQUESTED** [review]
Note that the size of the endpoint should be dynamic as a high-speed device can be connected as low-speed. Anyway that's a step in the right direction.

However for that reason we should not default the SAME70 controller to high speed in the DT, the device won't work properly if it is connected as low speed (see #11000 for more details). Leaving low speed as the default means the device will always work, albeit with lower throughput.

File: scripts/kconfig/kconfigfunctions.py [hunk 1]
```diff
@@ -388,6 +388,32 @@ def dt_node_int_prop(kconf, name, path, prop, unit=None):
     if name == "dt_node_int_prop_hex":
         return hex(_node_int_prop(node, prop, unit))
 
+def dt_node_prop_equal_str(kconf, _, path, prop, val):
```
Comment (mbolivar-nordic) [inline scripts/kconfig/kconfigfunctions.py:391]: I think for consistency with dt_node_int_prop_foo, this should be called dt_node_str_prop_equals or something like that.

**Review by mbolivar-nordic — COMMENTED** [review]
Looks good, just one bikeshed on naming the new kconfig function.

**Comment by jfischer-no** [comment]
> Note that the size of the endpoint should be dynamic as a high-speed device can be connected as low-speed. Anyway that's a step in the right direction.
>
> However for that reason we should not default the SAME70 controller to high speed in the DT, the device won't work properly if it is connected as low speed (see #11000 for more details). Leaving low speed as the default means the device will always work, albeit with lower throughput.

We do not support low-speed, you probably mean only full-speed capable host controller, but these days it is unlikely that anyone will have such a problem. But I will change option USB_DC_HAS_HS_SUPPORT so that one can disable it from menuconfig. Current device support implementation will remain such that either high-speed or full-speed are supported, not both at the same time.

File: scripts/kconfig/kconfigfunctions.py (hunk 1 above)
Comment (jfischer-no) [reply scripts/kconfig/kconfigfunctions.py]: Changed to dt_node_str_prop_equals.

**Review by mbolivar-nordic — APPROVED** [review]
For the DT portions

**Review by nandojve — APPROVED** [review]

**Comment by aurel32** [comment]
> > Note that the size of the endpoint should be dynamic as a high-speed device can be connected as low-speed. Anyway that's a step in the right direction.
> > However for that reason we should not default the SAME70 controller to high speed in the DT, the device won't work properly if it is connected as low speed (see #11000 for more details). Leaving low speed as the default means the device will always work, albeit with lower throughput.
> 
> We do not support low-speed, you probably mean only full-speed capable host controller, but these days it is unlikely that anyone will have such a problem. But I will change option USB_DC_HAS_HS_SUPPORT so that one can disable it from menuconfig. Current device support implementation will remain such that either high-speed or full-speed are supported, not both at the same time.

Sorry, I meant-full speed vs high-speed indeed. My point is that high-speed support is far from ready in Zephyr, and we should not enable it by default until it has been fully validated. Your patch series is a step in the right direction, but still are still missing, runtime speed selection (as opposed to build time) between full and high-speed mode is one thing, but there are more things like different meaning of the `bInterval` values for interrupt and isochronous transfers between full and high speed mode.

In short what I mean is that the `dts: same70: default USB HS controller to high-speed` commit should be dropped from this series. The other commits LGTM.

**Comment by jfischer-no** [comment]
>runtime speed selection (as opposed to build time) between full and high-speed mode is one thing

This is neither necessary nor possible in current USB device stack implementation, and again, current device support implementation will remain such that either high-speed or full-speed are supported, not both at the same time.

>but there are more things like different meaning of the bInterval values for interrupt and isochronous transfers between full and high speed mode.

Units of bInterval are different, 125 µs (microframe) or 1 ms (frame), but default values in the tree are still in valid range, even if they may not be suitable.

>In short what I mean is that the dts: same70: default USB HS controller to high-speed commit should be dropped from this series. The other commits LGTM.

The commit will stay, in case of trouble fallback is possible by disabling CONFIG_USB_DC_HAS_HS_SUPPORT.

**Review by carlescufi — APPROVED** [review]
